### PR TITLE
feat(aws-lambda): initial version of integration

### PR DESCRIPTION
**Pull request by scefali**
This PR sets up the initial version of the AWS Lambda Layer integration. This integration works by adding a layer to your Lambda functions that will initialize Sentry. [There is a separate PR to build and deploy the layer](https://github.com/getsentry/sentry-javascript/pull/3041). This way, a user can add Sentry instrumentation without updating their code at all! Note this is only targeting Node runtimes since we haven't found a way to bootstrap Sentry with Python. The integration is hidden behind the `organizations:integrations-aws_lambda` feature flag.

Here are the steps for how the integration works:
1. Press install which opens up a popup (shown below)
2. Click the link to create the CloudFormation stack
3. Paste the URL into the popup and press "submit".
4. Sentry queries all your Lambda functions and finds the Node runtimes
5. Sentry adds the layer and environment variables (`SENTRY_DSN`, `NODE_OPTIONS`, and `SENTRY_TRACES_SAMPLE_RATE`).
6. Installation is complete

Screen Shot 2020-12-07 at 2 45 04 PM

Note that this PR makes a number of shortcuts that will be addressed in the future:
1. We automatically choose the first Node serverless project we find and if there is none, we throw an error
2. We don't let the user pick which Lambdas they want to add Sentry to (in the future we will let them choose)
3. We obliterate any existing layers/env variables that exist for a Lambda function
4. We don't give the integration a unique name (hardcoded for now)
5. No unit tests (will add them in a future PR)

### CODE REVIEW

File: requirements-base.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 beautifulsoup4>=4.7.1,<4.8
-boto3>=1.4.1,<1.4.6
-botocore<1.5.71
+boto3>=1.10.1,<1.11
+botocore<1.14.0
```
Comment (scefali) [inline requirements-base.txt:3]: I have a separate PR to bump up the versions: https://github.com/getsentry/sentry/pull/22543

File: src/sentry/integrations/aws_lambda/integration.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            aws_external_id = request.POST["aws_external_id"]
+            pipeline.bind_state("arn", arn)
+            pipeline.bind_state("aws_external_id", aws_external_id)
+            return pipeline.next_step()
+
+        template_url = options.get("aws-lambda.cloudformation-url")
+        aws_external_id = uuid.uuid4()
+
+        cloudformation_url = (
+            "https://console.aws.amazon.com/cloudformation/home#/stacks/create/review?"
```
Comment (NisanthanNanthakumar) [inline src/sentry/integrations/aws_lambda/integration.py:104]: @scefali now that we are on python3 we can use f-strings. %-formatting is not recommended for new code.

File: src/sentry/integrations/aws_lambda/utils.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from __future__ import absolute_import
+
+
+# Taken from: https://gist.github.com/gene1wood/5299969edc4ef21d8efcfea52158dd40
+def parse_arn(arn):
```
Comment (NisanthanNanthakumar) [inline src/sentry/integrations/aws_lambda/utils.py:5]: @scefali we should make tests for this function

**Review by NisanthanNanthakumar — APPROVED** [review]

File: src/sentry/integrations/aws_lambda/utils.py (hunk 3 above)
Comment (scefali) [reply src/sentry/integrations/aws_lambda/utils.py:5]: @NisanthanNanthakumar for sure, i'm going to add tests in a future PR

File: src/sentry/integrations/aws_lambda/integration.py (hunk 2 above)
Comment (scefali) [reply src/sentry/integrations/aws_lambda/integration.py:104]: @NisanthanNanthakumar so using `.format` you mean?

**Pull request merged**